pydis_site/apps/content/utils.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+from typing import Dict, Union
+
+import yaml
+from django.http import Http404
+from markdown2 import markdown
+
+
+def get_category(path: Path) -> Dict[str, str]:
+    """Load category information by name from _info.yml."""
+    if not path.exists() or not path.is_dir():
+        raise Http404("Category not found.")
+
+    return yaml.safe_load(path.joinpath("_info.yml").read_text())
+
+
+def get_categories(path: Path) -> Dict[str, Dict]:
+    """Get all categories information."""
+    categories = {}
+
+    for name in path.iterdir():
+        if name.is_dir():
+            categories[name.name] = get_category(path.joinpath(name.name))
+
+    return categories
+
+
+def get_pages(path: Path) -> Dict[str, Dict]:
+    """Get all root or category page names and their metadata."""
```

Comment:
The phrasing of the docstring is a bit confusing. If I understand correctly, it's mapping the names of pages in a category at `path` to their metadata. Perhaps `get_category_pages` is a more appropriate function name.